cdap-app-fabric/src/main/java/co/cask/cdap/route/store/ZKRouteStore.java

```diff
@@ -120,11 +121,9 @@ public RouteConfig fetch(final ProgramId serviceId) {
   private RouteConfig getConfig(ProgramId serviceId, Future<RouteConfig> future) {
```

Comment:
Better yet, is not to return `null` from this map, but rather have the logic all an empty `RouteConfig` (basically an empty map). Whenever there is an empty map, fallback to default routing strategy.